fancy_collections/formatting.py

```diff
@@ -45,25 +47,49 @@ def key_to_string(self, key: Any, obj: Any) -> str:  # noqa
     def to_string(self) -> str:
         if len(self._obj) == 0:
             return self._stringify_empty_class(self._obj)
+
+        objects = {}
+        widths = {}
         for key, val in self._obj.items():
             key = self.key_to_string(key, val)
-            try:
-                string = self.stringify(val)
-            except Exception:  # noqa
-                string = NotImplemented
-            if string is NotImplemented:
-                string = str(val)
-            self._add(key, string.splitlines())
+            lines = self.stringify(val).splitlines()
+            objects[key] = lines
+            widths[key] = self._get_maxlen(lines + [key])
+
+        display_width = int(shutil.get_terminal_size().columns * 0.8)
```

Comment:
why this `* 0.8` ?

Reply:
Well, try and error :grimacing: .  As the rendering seems to add a few characters here and there, going with the full terminal size still resulted in line breaks. The factor 0.8 seemed to work very well in my testing.

Reply:
i'm a bit unhappy with this _guess_ [1]. 
IMO it the total length is deterministic and can be calculated like so :

`self._get_maxlen(lines + [key]) + column seperator` for each column except the last does not need the seperator.

so something like this pseudocode should do `sum([widths[k] for k in widths.keys()]) + len(sep) * (len(widths)-1)`    


-----------
[1] 
>  In the face of ambiguity, refuse the temptation to guess.

from pep20...

Reply:
We also have:

> Simple is better than complex.

> Readability counts.

> Although practicality beats purity.

Reply:
Changed...